torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1704,8 +1712,12 @@ c10::List<Elem> makeListForDictKeysOrValues(
 
 template <unsigned int Index>
 c10::impl::GenericList makeGenericListForDictKeysOrValues(
+    const std::pair<c10::optional<TypePtr>, c10::optional<TypePtr>>& types,
     const std::vector<std::pair<IValue, IValue>>& order) {
-  auto values = c10::impl::GenericList(c10::impl::deprecatedUntypedList());
+  auto type = std::get<Index>(types);
```

Comment:
same here

Reply:
Here, it's more than just error checking. Without this, the runtime key/value type of the Dict wouldn't propagate into the resulting list.

Reply:
We create the type of the list based on the static type. E.g. an optional list of ints with a value of [1] should be a generic list. If it this created an intlist that would be a bug. 

how exactly is the runtime type set ? and as with ^^^ the type system should already enforce the correct runtime value